prelude/haskell/haskell.bzl

```diff
@@ -536,7 +538,8 @@ def _build_haskell_lib(
     lib_short_path = paths.join("lib-{}".format(artifact_suffix), libfile)
 
     linfos = [x.prof_info if enable_profiling else x.info for x in hlis]
-    uniq_infos = dedupe(flatten([x[link_style] for x in linfos]))
+    # only gather direct dependencies
+    uniq_infos = [x[link_style].reduce("root") for x in linfos]
```

Comment:
We just noticed that there is a `value` field available on a tset. This would render the `root` reduction useless. Should I change this and push again, or is there any reason not to use `value`?

Reply:
That sounds like a great plan. I've just landed it, so a new diff would be most welcome.

Reply:
PR is up: https://github.com/facebook/buck2/pull/621